src/main/java/net/javadiscord/javabot/systems/commands/ChangeMyMindCommand.java

```diff
@@ -24,14 +24,26 @@
  */
 @Deprecated
 public class ChangeMyMindCommand implements SlashCommand {
+	/**
+	 * The maximum acceptable length for texts to send to the API. Technically,
+	 * the API supports up to but not including 2000, but we'll make that much
+	 * lower to avoid issues, since the API fails with a 501 error. People
+	 * shouldn't really be putting paragraphs into this anyway.
```

Comment:
Are you sure that 1500 characters is the correct length? What about unicode letters with multiple codepoints?

Reply:
There is no documented length. So I don't know, but the issue said that 2000 characters breaks it. So limiting much lower is just an attempt at making it work in all cases.

Reply:
Have you tested whether ot works with 1500 characters?

Reply:
Yeah apparently 1500 is still too much, so I changed to 500 which does work.

Reply:
Do unicode characters with multiple codepoints (like emojis) change anything?

Reply:
Well, I did this test:
```java
class Scratch {
	public static void main(String[] args) {
		String text = "\uD83D\uDE00".repeat(250);
		String encodedSearchTerm;
		encodedSearchTerm = URLEncoder.encode(text, StandardCharsets.UTF_8);
		System.out.println(encodedSearchTerm.length());
		System.out.println(encodedSearchTerm.toCharArray().length);
		System.out.println(encodedSearchTerm.getBytes(StandardCharsets.UTF_8).length);
	}
}
```
Which shows that my logic works correctly to determine the actual length of a string with many multiple codepoint characters, where it outputs `3000` as length for all cases.

So no, emojis don't change anything; it still works.

Reply:
Can you try a binary search or similar for finding out tze maximum length?

Reply:
Well, I'm pretty sure it's just a limitation due to the length of the query string. But it's pointless to try and find an exact max since with 500 characters the text already goes off the image.